src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php

```diff
@@ -209,6 +195,135 @@ public function isInitializable(string $class, string $property, array $context
+            $method = $reflClass->getMethod($getsetter);
+
+            return PropertyReadInfo::forGetter($getsetter, $this->getReadVisiblityForMethod($method), $method->isStatic());
+        }
+
+        if ($hasProperty && ($reflClass->getProperty($property)->getModifiers() & $this->propertyReflectionFlags)) {
+            $reflProperty = $reflClass->getProperty($property);
+
+            return PropertyReadInfo::forProperty($property, $this->getReadVisiblityForProperty($reflProperty), $reflProperty->isStatic(), true);
+        }
+
+        if ($reflClass->hasMethod('__get') && ($reflClass->getMethod('__get')->getModifiers() & $this->methodReflectionFlags)) {
+            return PropertyReadInfo::forProperty($property, PropertyReadInfo::VISIBILITY_PUBLIC, false, false);
+        }
+
+        if ($allowMagicCall && $reflClass->hasMethod('__call') && ($reflClass->getMethod('__call')->getModifiers() & $this->methodReflectionFlags)) {
+            return PropertyReadInfo::forGetter('get'.$camelProp, PropertyReadInfo::VISIBILITY_PUBLIC, false);
+        }
+
+        return null;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getWriteInfo(string $class, string $property, array $context = []): ?PropertyWriteInfo
+    {
+        try {
+            $reflClass = new \ReflectionClass($class);
+        } catch (\ReflectionException $e) {
+            return null;
+        }
+
+        $allowGetterSetter = $context['enable_getter_setter_extraction'] ?? false;
+        $allowMagicCall = $context['enable_magic_call_extraction'] ?? false;
+        $allowConstruct = $context['enable_constructor_extraction'] ?? $this->enableConstructorExtraction;
+        $allowAdderRemover = $context['enable_adder_remover_extraction'] ?? true;
+
+        $camelized = $this->camelize($property);
+        $constructor = $reflClass->getConstructor();
+
+        if (null !== $constructor && $allowConstruct) {
+            foreach ($constructor->getParameters() as $parameter) {
+                if ($parameter->getName() === $property) {
+                    return PropertyWriteInfo::forConstructor($property);
+                }
+            }
+        }
+
+        if ($allowAdderRemover && null !== $methods = $this->findAdderAndRemover($reflClass, (array) Inflector::singularize($camelized))) {
+            [$adderAccessName, $removerAccessName] = $methods;
+
+            $adderMethod = $reflClass->getMethod($adderAccessName);
+            $removerMethod = $reflClass->getMethod($removerAccessName);
+
+            return PropertyWriteInfo::forAdderAndRemover(
+                PropertyWriteInfo::forSetter($adderAccessName, $this->getWriteVisiblityForMethod($adderMethod), $adderMethod->isStatic()),
+                PropertyWriteInfo::forSetter($removerAccessName, $this->getWriteVisiblityForMethod($removerMethod), $removerMethod->isStatic())
```

Comment:
I understand now why this was called "method" and not "setter" before.

* A write info of type "method" is an arbitrary method
* A write info of type "adder and remover" is a composition of two "method" write infos
* A write info of type "constructor" is simplified and does not bear a method name because the method name is implicit

Given that, I think "method" was indeed a better choice of naming than "setter" and "getter". So I think reverting this to `forMethod()` and `TYPE_METHOD` (same for read info) makes sense.

Sorry for the back and forward. :)

Reply:
Fixed